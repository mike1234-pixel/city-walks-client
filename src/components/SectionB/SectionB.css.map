{"version":3,"sourceRoot":"","sources":["SectionB.scss"],"names":[],"mappings":"AAAA;AAEA;EACE;EACA;;;AAGF;EACE;IACE;IACA;IACA;IACA;IACA;;;AAIJ;EACE;IACE;IACA;;EAGF;IACE;IACA;;EAGF;IACE;IACA;;;AAIJ;EACE;EACA;EACA;EACA;EACA;EACA;EACA;;;AAGF;AAAA;EAEE;EACA;EACA;;;AAGF;EACE;EACA;EACA;EACA;;AAEA;EAGE;;;AAIJ;EACE;EACA","file":"SectionB.css"}