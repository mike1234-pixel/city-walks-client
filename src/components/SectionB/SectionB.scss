/** @format */

.section-b {
  width: 100%;
  max-width: 100%;
}

@media (min-width: 991px) {
  .grid-container-b {
    display: grid;
    grid-template-columns: repeat(2, 1fr);
    grid-template-rows: 100%;
    grid-gap: 30px;
    padding: 30px 0;
  }
}

@media (max-width: 992px) {
  .grid-container-b {
    display: flex;
    flex-direction: column;
  }

  .grid-item-b-text-box {
    margin-bottom: 30px;
    order: 0;
  }

  .grid-item-b-image-box {
    margin-bottom: 30px;
    order: 1;
  }
}

.grid-item-b {
  display: flex;
  align-items: center;
  justify-content: center;
  flex-direction: column;
  transition: all 0.4s;
  box-sizing: border-box;
  border: 2px solid transparent;
}

.grid-item-b:hover,
.grid-item-b:active {
  background: transparent;
  color: var(--dark);
  cursor: pointer;
}

.grid-item-b-text-box {
  padding: 20px;
  text-align: left;
  background: var(--dark);
  color: white;

  &:hover,
  &:focus,
  &:active {
    border: 2px dashed grey;
  }
}

.grid-item-b-img {
  width: 100%;
  height: 100%;
}