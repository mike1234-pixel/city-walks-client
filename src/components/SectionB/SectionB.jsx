import { MDBView } from "mdbreact"
import marked from "marked"
import "./SectionB.css";

const SectionB = (props) => {

    const {content, img, alt} = props

    const createMarkup = (markup) => {
      return {__html: marked(markup)}
    }

  return (
    <div className='section-b grid-container-b'>
        <div className='grid-item-b grid-item-b-image-box'>
          <MDBView hover zoom>
            <img className="grid-item-b-img" src={img} alt={alt}></img>
          </MDBView>
        </div>
        <div className='grid-item-b grid-item-b-text-box'>
            <div  dangerouslySetInnerHTML={createMarkup(content)}></div>
        </div>
    </div>
  );
};

export default SectionB