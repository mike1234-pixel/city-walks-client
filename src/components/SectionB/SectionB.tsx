import React from 'react'
import { MDBView } from "mdbreact"
import marked from "marked"
import "./SectionB.scss"

interface Props {
  content: string;
  img: string;
  alt: string;
}

const SectionB: React.FC<Props> = (props: Props) => {

  const { content, img, alt } = props

  const createMarkup = (markup: string) => {
    return { __html: marked(markup) }
  }

  return (
    <div className='section-b grid-container-b'>
      <div className='grid-item-b grid-item-b-image-box'>
        <MDBView hover zoom id="grid-item-overlay" waves>
          <img className="grid-item-b-img" src={img} alt={alt}></img>
        </MDBView>
      </div>
      <div className='grid-item-b grid-item-b-text-box'>
        <div dangerouslySetInnerHTML={createMarkup(content)}></div>
      </div>
    </div>
  );
};

export default SectionB