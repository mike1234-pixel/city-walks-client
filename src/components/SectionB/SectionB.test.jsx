import React from 'react'
import renderer from 'react-test-renderer'
import SectionB from './SectionB'

// snapshot test
it('Section B renders correctly', () => {  
    
    const content = "mock content"
    const img = "imagefile"
    const alt = "alt attribute"

  const tree = renderer
    .create(
        <SectionB content={content} img={img} alt={alt}/>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});