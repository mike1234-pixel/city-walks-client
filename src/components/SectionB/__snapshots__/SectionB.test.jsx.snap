// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Section B renders correctly 1`] = `
<div
  className="section-b grid-container-b"
>
  <div
    className="grid-item-b grid-item-b-image-box"
  >
    <div
      className="view overlay zoom"
      data-test="view"
      onMouseDown={[Function]}
      onTouchStart={[Function]}
      style={Object {}}
    >
      <img
        alt="alt attribute"
        className="grid-item-b-img"
        src="imagefile"
      />
    </div>
  </div>
  <div
    className="grid-item-b grid-item-b-text-box"
  >
    <p>
      mock content
    </p>
  </div>
</div>
`;
